src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -126,24 +128,27 @@ trait RegularSync {
         //we have same chain prefix
         if (parent.hash == headers.head.parentHash) {
 
-          val currentBranchTotalDifficulty = headersQueue.map(_.number)
+          val oldBlocks: Seq[Option[(BlockBody, BigInt)]] = headersQueue.map(_.number)
```

Comment:
`Seq[(BlockBody, BigInt)]`? It's seems we don't have any interest in those Nones.
Also, if I'm reading this correctly, blocks in `headersQueue` should be in order. So shouldn't we give up after first `getBlockByNumber` that returns `None`?